### ISSUE TITLE
instalacja

### ISSUE THREAD

**Issue by bogut82** [MichalKrasowski/tech-controllers#8]
Cześć,

wybacznie za lamerstwo ale mam problem z instalacją tego dodatku. po skopiowaniu plików do katalogu nic nie widać. druga metoda przez hacs działa połownicznie tj. dodatek jest widoczny:

![image](https://user-images.githubusercontent.com/21171398/176994795-1bb23867-08a4-4f72-99b3-ba2936d2b337.png)

ale w integracjach już mi tego nie znajduje:

![image](https://user-images.githubusercontent.com/21171398/176994859-d369b478-da3f-4f39-9e15-adade5bf5c63.png)

Dopiero zaczynam swoją przygodę z HA więc mogę czegoś jeszcze nie wiedzieć.

z góry dzięki za pomoc, Łukasz

**Comment by djgprv** [comment]
Niestety mam tak samo:( Najnowsza dostępna wersja HA.
W zeszłym tygodniu wymieniony zostal rekuperator ze sterownikiem tech. Szkoda, że projekt nie jest rozwijany. Pozdrawiam

**Comment by xeolcz** [comment]
I have the same problem, any solution?

**Comment by djgprv** [comment]
> I have the same problem, any solution?

Try this one fork: https://github.com/souzen/tech-controllers. Works for me (installation).

**Comment by Kacper2Kaczmarek** [comment]
> > I have the same problem, any solution?
> 
> Try this one fork: https://github.com/souzen/tech-controllers. Works for me (installation).

Is it still working for you? I'm trying to install it and it's not showing up in the Add integrations menu

**Comment by djgprv** [comment]
Yes, it is but ... my module is not supported Karino ST-340 v.1.0.22
(recuperator) so i can't see any entities. Just active integration :(

wt., 4 paź 2022 o 17:45 Kacper2Kaczmarek ***@***.***>
napisał(a):

**Comment by djgprv** [comment]
Can someone help me add support for Tech controler Karino ST-340 v.1.0.22 (recuperator)? I would like to be able to read the temperature, change the fan speed, turn on, turn off the bypass and airing.

**Comment by Zbiku91** [comment]
Same here... controller WANAS: ST-340 V2 (v.2.0.11), added to the HA but no entity available. 
I was able to get the API response from the driver's api but probably this needs to be implemented...
FYI, this is the JSON response:
`{
	"zones": {
		"transaction_time": null,
		"elements": [],
		"globalSchedules": {
			"time": null,
			"duringChange": null,
			"elements": []
		},
		"controllerParameters": {}
	},
	"tiles": [
		{
			"id": 5509,
			"parentId": 0,
			"type": 50,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Controller software version",
				"version": "2.0.11",
				"txtId": 1634,
				"iconId": 29,
				"companyId": 23,
				"controllerName": "ST-340 V2",
				"mainControllerId": 93
			}
		},
		{
			"id": 5761,
			"parentId": 0,
			"type": 1,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Temperature sensor",
				"workingStatus": true,
				"txtId": 795,
				"value": 110,
				"batteryLevel": null,
				"signalStrength": null
			}
		},
		{
			"id": 5785,
			"parentId": 0,
			"type": 1,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Temperature sensor",
				"workingStatus": true,
				"txtId": 1841,
				"value": 132,
				"batteryLevel": null,
				"signalStrength": null
			}
		},
		{
			"id": 6038,
			"parentId": 0,
			"type": 1,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Temperature sensor",
				"workingStatus": true,
				"txtId": 1840,
				"value": 131,
				"batteryLevel": null,
				"signalStrength": null
			}
		},
		{
			"id": 6039,
			"parentId": 0,
			"type": 1,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Temperature sensor",
				"workingStatus": true,
				"txtId": 1842,
				"value": 107,
				"batteryLevel": null,
				"signalStrength": null
			}
		},
		{
			"id": 6050,
			"parentId": 0,
			"type": 11,
			"menuId": 1049,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Relay",
				"workingStatus": false,
				"txtId": 1814,
				"iconId": 123
			}
		},
		{
			"id": 6051,
			"parentId": 0,
			"type": 6,
			"menuId": 1051,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Universal status with widgets",
				"statusId": 1,
				"iconId": 86,
				"widget1": {
					"txtId": 0,
					"value": 0,
					"unit": -1,
					"type": 0,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				},
				"widget2": {
					"txtId": 5756,
					"value": 0,
					"unit": 14,
					"type": 1,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				}
			}
		},
		{
			"id": 6053,
			"parentId": 0,
			"type": 6,
			"menuId": 1053,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Universal status with widgets",
				"statusId": 0,
				"iconId": 5,
				"widget1": {
					"txtId": 0,
					"value": 0,
					"unit": -1,
					"type": 0,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				},
				"widget2": {
					"txtId": 1843,
					"value": 0,
					"unit": 16,
					"type": 1,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				}
			}
		},
		{
			"id": 6054,
			"parentId": 0,
			"type": 6,
			"menuId": 1054,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Universal status with widgets",
				"statusId": 0,
				"iconId": 146,
				"widget1": {
					"txtId": 0,
					"value": 0,
					"unit": -1,
					"type": 0,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				},
				"widget2": {
					"txtId": 1844,
					"value": 0,
					"unit": 17,
					"type": 1,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				}
			}
		},
		{
			"id": 6073,
			"parentId": 0,
			"type": 41,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Date",
				"year": 2023,
				"month": 4,
				"day": 19,
				"dayId": 3,
				"hours": 8,
				"minutes": 33
			}
		},
		{
			"id": 6341,
			"parentId": 0,
			"type": 6,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Universal status with widgets",
				"statusId": 0,
				"iconId": 102,
				"widget1": {
					"txtId": 0,
					"value": 0,
					"unit": -1,
					"type": 0,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				},
				"widget2": {
					"txtId": 6158,
					"value": 12,
					"unit": 0,
					"type": 1,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				}
			}
		},
		{
			"id": 6342,
			"parentId": 0,
			"type": 6,
			"menuId": 0,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Universal status with widgets",
				"statusId": 0,
				"iconId": 102,
				"widget1": {
					"txtId": 0,
					"value": 0,
					"unit": -1,
					"type": 0,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				},
				"widget2": {
					"txtId": 6157,
					"value": 21,
					"unit": 0,
					"type": 1,
					"params": [
						0,
						0,
						0,
						0,
						0
					]
				}
			}
		},
		{
			"id": 6445,
			"parentId": 0,
			"type": 11,
			"menuId": 1445,
			"orderId": null,
			"visibility": true,
			"params": {
				"description": "Relay",
				"workingStatus": false,
				"txtId": 1844,
				"iconId": 123
			}
		}
	]
}`

**Comment by djg-io** [comment]
Modbu isn't working for me too:/ Anyone have an example of reading data from st-340 using modbus?
